pyro/infer/svi.py

```diff
@@ -51,6 +59,19 @@ def _loss_and_grads(*args, **kwargs):
             self.loss = loss
             self.loss_and_grads = loss_and_grads
 
+    def run(self, *args, **kwargs):
+        if self.num_steps > 0:
+            with poutine.block():
+                for i in range(self.num_steps):
+                    self.step(*args, **kwargs)
+        return super(SVI, self).run(*args, **kwargs)
+
+    def _traces(self, *args, **kwargs):
+        for i in range(self.num_samples):
+            guide_trace = poutine.trace(self.guide).get_trace(*args, **kwargs)
+            model_trace = poutine.trace(poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
+            yield model_trace, (model_trace.log_prob_sum() - guide_trace.log_prob_sum())
```

Comment:
We can probably set the importance weights to some constant here, right (seems like double adjustment otherwise)?

Reply:
Yeah, you're right